pinot-controller/src/main/java/org/apache/pinot/controller/recommender/rules/AbstractRule.java

```diff
@@ -16,22 +16,19 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.pinot.tools.tuner.query.src.parser;
+package org.apache.pinot.controller.recommender.rules;
 
-import javax.annotation.Nullable;
-import org.apache.pinot.tools.tuner.query.src.stats.wrapper.AbstractQueryStats;
+import org.apache.pinot.controller.recommender.io.ConfigManager;
+import org.apache.pinot.controller.recommender.io.InputManager;
 
 
-/**
- * Parser interface for a query line
- */
-public interface QueryParser {
-  /**
-   * parse the the complete log line to a parsed obj
-   * @param line the complete log line to be parsed, InputIterator should put broken lines together
-   * @return the parsed log line obj
-   */
-  @Nullable
-  AbstractQueryStats parse(String line);
-}
+public abstract class AbstractRule {
```

Comment:
This should be a new class/interface for the recommender right? Why are we moving an existing class? or is this a github issue?

Reply:
Please see the above conversation.

Reply:
Got it. They came from tuner